design/fips-compliance.md

```diff
@@ -0,0 +1,199 @@
+
+### Non-FIPS compliant image
+
+We should decide if we still want to release a version of the `conjur-authn-k8s-client`
+that is not FIPS compliant (i.e that uses `golang` as the base image). Let's
+look into some pros and cons of both approaches.
+ 
+ | **Option**                             | **Implications**                                                                                                                                                                                                                                                                                                                                                        |
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that Releasing both
+versions will bring to the development team.
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
+
+## Test plan
+
+Before we dig into which tests should run, we should decide where, and how, they
+should run. Currently, the `conjur-authn-k8s-client` have UTs in its project
+and a vanilla test that runs in `kubernetes-conjur-demo`. The `kubernetes-conjur-demo`
```

Comment:
Why is a vanilla test in the `kubernetes-conjur-demo` repo? I am not sure what this gives us but it sounds a bit off for us to have a core test in another repo? Can we consider moving this test over to `conjur-authn-k8s-client`?

Reply:
Well, this is the current flow that we have. There are pros and cons for that. 

Option 3 moves the tests into our own repo.

Reply:
aware of why this decision was made?

Reply:
Nope. I am guessing they wanted some centralized place to run tests so the builds of the authn-client and secretless are faster